Roles/Impostor/DollMaster.cs

```diff
@@ -0,0 +1,335 @@
+    private const int Id = 28500;
+    private static readonly HashSet<byte> PlayerIds = [];
+    public static bool HasEnabled => PlayerIds.Any();
+    public override bool IsExperimental => true;
+    public override bool IsEnable => HasEnabled;
+    public override CustomRoles ThisRoleBase => CustomRoles.Shapeshifter;
+    public override Custom_RoleType ThisRoleType => Custom_RoleType.ImpostorKilling;
+    //==================================================================\\
+
+    private static OptionItem DefaultKillCooldown;
+    private static OptionItem ShapeshiftCooldown;
+    private static OptionItem ShapeshiftDuration;
+    public static OptionItem CanKillAsMainBody;
+    public static OptionItem TargetDiesAfterPossession;
+
+    public override void SetupCustomOption()
+    {
+        SetupRoleOptions(Id, TabGroup.OtherRoles, CustomRoles.DollMaster);
+        DefaultKillCooldown = FloatOptionItem.Create(Id + 10, "KillCooldown", new(0f, 180f, 2.5f), 25f, TabGroup.OtherRoles, false).SetParent(CustomRoleSpawnChances[CustomRoles.DollMaster])
+            .SetValueFormat(OptionFormat.Seconds);
+        ShapeshiftCooldown = FloatOptionItem.Create(Id + 11, "DollMasterPossessionCooldown", new(0f, 180f, 2.5f), 25f, TabGroup.OtherRoles, false).SetParent(CustomRoleSpawnChances[CustomRoles.DollMaster])
+            .SetValueFormat(OptionFormat.Seconds);
+        ShapeshiftDuration = FloatOptionItem.Create(Id + 12, "DollMasterPossessionDuration", new(0f, 180f, 2.5f), 10f, TabGroup.OtherRoles, false).SetParent(CustomRoleSpawnChances[CustomRoles.DollMaster])
+            .SetValueFormat(OptionFormat.Seconds);
+        CanKillAsMainBody = BooleanOptionItem.Create(Id + 13, "DollMasterCanKillAsMainBody", true, TabGroup.OtherRoles, false).SetParent(Options.CustomRoleSpawnChances[CustomRoles.DollMaster]);
+        TargetDiesAfterPossession = BooleanOptionItem.Create(Id + 14, "DollMasterTargetDiesAfterPossession", false, TabGroup.OtherRoles, false).SetParent(Options.CustomRoleSpawnChances[CustomRoles.DollMaster]);
+    }
+
+    public override void Init()
+    {
+        ReducedVisionPlayers.Clear();
+        PlayerIds.Clear();
+        isEnabled = true;
+    }
+
+    public override void Add(byte playerId)
+    {
+        PlayerIds.Add(playerId);
+        DollMasterTarget = Utils.GetPlayerById(playerId);
+        IsControllingPlayer = false;
+    }
+
+    public override void ApplyGameOptions(IGameOptions opt, byte playerId)
+    {
+        AURoleOptions.ShapeshifterCooldown = ShapeshiftCooldown.GetFloat();
+        AURoleOptions.ShapeshifterDuration = ShapeshiftDuration.GetFloat() + 0.35f;
+    }
+
+    public override void SetKillCooldown(byte id) => Main.AllPlayerKillCooldown[id] = DefaultKillCooldown.GetFloat();
+
+    // A quick check if a player is being possessed.
+    public static bool IsDoll(byte id)
+    {
+        if (ReducedVisionPlayers.Contains(id))
+            return true;
+        return false;
+    }

```

Comment:
Here you can simplify the code by replacing it with:
`public static bool IsDoll(byte id) => ReducedVisionPlayers.Contains(id);`

Reply:
![image](https://github.com/0xDrMoe/TownofHost-Enhanced/assets/143632280/4da84354-ea47-43be-9992-f49657e50120)
same here